pyproject.toml

```diff
@@ -2,7 +2,7 @@
 requires = [
     "setuptools>=42",
     "setuptools_scm[toml]>=3.4",
-    "pybind11>=2.6.1",
+    "pybind11>=2.12.0",
```

Comment:
I guess this isn't really necessary(?) to bump it given that there's already a lower bound here, but I guess signaling this also doesn't hurt.